admin/category_product_listing.php

```diff
@@ -816,17 +820,17 @@
 // Split Page
 
 // reset page when page is unknown
-            if ((empty($_GET['page']) || $_GET['page'] == '1') && !empty($_GET['pID'])) {
+            if ((empty($_GET['page']) || $_GET['page'] === 1) && !empty($_GET['pID'])) {
               $check_page = $db->Execute($products_query_raw);
               if ($check_page->RecordCount() > $max_results) {
                 $check_count = 0;
                 foreach ($check_page as $item) {
-                  if ($item['products_id'] == $_GET['pID']) {
+                  if ($item['products_id'] === $_GET['pID']) {
                     break;
                   }
                   $check_count++;
                 }
-                $_GET['page'] = round((($check_count / $max_results) + (fmod_round($check_count, $max_results) != 0 ? .5 : 0)), 0);
+                $_GET['page'] = round((($check_count / $max_results) + (fmod_round($check_count, $max_results) !== 0 ? .5 : 0)));
```

Comment:
This one's tough, since `fmod_round` is going to return either an int or a float; I'll suggest leaving the comparison as a lazy one.